### DISCUSSION TITLE
Doesn't rise native .net exception

### DISCUSSION THREAD

**Discussion by akarpuhov** [fluentassertions/fluentassertions#1780] [Q&A] [answered]
If I use this code only `AssertionException` raises:

```cs
using (new AssertionScope())
{
    AssertionScope.Current.FailWith("AssertionException");
    throw new Exception("Exception");
}
```

How I can catch .net exception ?

**Comment by dennisdoomen** [comment]
I'm not sure what you're trying to do. `AssertionScope` has nothing to do with throwing .NET exceptions.

**Reply by akarpuhov** [reply]
If you try this code .net exception raises. But why it doesn't raise in first case ?
```
using (new AssertionScope())
{
    throw new Exception("Exception");
    AssertionScope.Current.FailWith("AssertionException");
}
```

**Comment by dennisdoomen** [comment]
Ah, you mean that in the first example, you were not expecting the `AssertionScope.Current.FailWith("AssertionException");` to throw?

**Comment by dennisdoomen** [comment, marked as answer]
Actually, it works as expected. The `AssertionException` isn't thrown until the `using` block ends. The problem is that the `Dispose` of `AssertionScope` throws its own exception and that hides the other exception that was thrown in the same scope.
